core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/suggest/TermConceptSuggester.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2022 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.core.request.suggest;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import org.tartarus.snowball.ext.EnglishStemmer;
+
+import com.b2international.commons.http.ExtendedLocale;
+import com.b2international.index.compat.TextConstants;
+import com.b2international.snomed.ecl.Ecl;
+import com.b2international.snowowl.core.codesystem.CodeSystemRequests;
+import com.b2international.snowowl.core.events.util.Promise;
+import com.b2international.snowowl.core.plugin.Component;
+import com.b2international.snowowl.core.request.SearchIndexResourceRequest;
+import com.b2international.snowowl.core.request.search.TermFilter;
+import com.b2international.snowowl.eventbus.IEventBus;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Splitter;
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Multiset;
+import com.google.common.collect.Multisets;
+
+/**
+ * @since 8.5
+ */
+@Component
+@JsonTypeName("term")
+public final class TermConceptSuggester implements ConceptSuggester {
+
+	// Split terms at delimiter or whitespace separators
+	private static final Splitter TOKEN_SPLITTER = Splitter.on(TextConstants.WHITESPACE_OR_DELIMITER_MATCHER)
+			.trimResults()
+			.omitEmptyStrings();
+	
+	/**
+	 * The number of most frequent tokens (words) to consider from the selected like text corpus. Default value is 9.
+	 */
+	@JsonProperty
+	private Integer topTokenCount = 9;
+	
+	/**
+	 * The number of most frequent tokens (topTokenCount) to match in order to be accepted as a suggestion. Default value is 3. 
+	 */
+	@JsonProperty
+	private Integer minOccurenceCount = 3;
+	
+	/**
+	 * The minimum length for a token to be considered as potential top token. Default value is 0, accepting all token lengths.
+	 */
+	@JsonProperty
+	private int minTokenLength = 0;
+	
+	/**
+	 * Whether to ignore any stopwords in the selected text corpus or not. Default value is <code>true</code>.
+	 */
+	@JsonProperty
+	private boolean ignoreStopwords = true;
+	
+	/**
+	 * Whether to perform english word stemming on the tokens or not. Default value is false.
+	 */
+	@JsonProperty
+	private boolean stemming = false;
```

Comment:
The previous default was to remove affixes from the end of word stems, in other words `true`. We might want to keep it for backwards compatibility.

Reply:
Yeah, this is not meant to be fully backward compatible and the former API was in an experimental state, so this is fine. Stemming was only used by default when you have selected a range of concepts via a query. It was impossible to request stemming when you were only passing in a `term` parameter. This now allows the caller to decide whether the basis of the suggestion should be stemmed or not, regardless of where the source terms came from (defined externally or selected via ECL).